apps/devhub/tests/test_views_versions.py

```diff
@@ -227,30 +227,69 @@ def test_non_owner_cant_enable_addon(self):
         eq_(res.status_code, 403)
         eq_(Addon.objects.get(id=3615).disabled_by_user, False)
 
-    def test_show_disable_button(self):
+    def test_non_owner_cant_change_status(self):
+        """A non-owner can't use the radio buttons."""
         self.addon.update(disabled_by_user=False)
+        addon_user = AddonUser.objects.get(addon=self.addon)
+        addon_user.role = amo.AUTHOR_ROLE_VIEWER
+        addon_user.save()
         res = self.client.get(self.url)
         doc = pq(res.content)
+        assert doc('.enable-addon').attr('checked') == 'checked'
+        assert doc('.enable-addon').attr('disabled') == 'disabled'
+        assert not doc('.disable-addon').attr('checked')
+        assert doc('.disable-addon').attr('disabled') == 'disabled'
+        assert not doc('.unlist-addon').attr('checked')
+        assert doc('.unlist-addon').attr('disabled') == 'disabled'  # Readonly.
+
+    def test_published_addon_radio(self):
+        """Published (listed) addon is selected: can hide or publish."""
+        self.addon.update(disabled_by_user=False)
+        res = self.client.get(self.url)
+        doc = pq(res.content)
+        assert doc('.enable-addon').attr('checked') == 'checked'
+        assert not doc('.enable-addon').attr('disabled')
```

Comment:
Add a check for `data-url` to help make sure the JS is going to work, since we don't have front-end tests ?